src/v2/components/Search/index.jsx

```diff
@@ -1,40 +1,79 @@
 // @flow
 import React, {useState} from 'react';
+import {observer} from 'mobx-react-lite';
+import {compose, get, filter, lowerCase, contains} from 'lodash/fp';
 import {InputBase, IconButton} from '@material-ui/core';
 import {Search as SearchIcon} from '@material-ui/icons';
+import NodesStore from 'v2/stores/nodes';
 
+import SearchResult from './result';
 import useStyles from './styles';
 
-type SearchProps = {
-  onSubmit: (val: string) => void,
-};
-
-const Search = ({onSubmit}: SearchProps) => {
+const Search = () => {
   const classes = useStyles();
+  const {validators} = NodesStore;
   const [value, setValue] = useState('');
+  const [isDirty, setDirty] = useState(false);
+  const [isFocus, setFocus] = useState(false);
+  const [searchResult, setSearchResult] = useState([]);
+
+  const onFocus = () => setFocus(true);
+  const onBlur = () => {
+    setTimeout(() => {
+      setFocus(false);
+    }, 250);
+  };
+
+  const handleClear = () => {
+    setDirty(false);
+    setValue('');
+    setSearchResult([]);
+  };
+
   const handleSearch = ({currentTarget}: SyntheticEvent<HTMLInputElement>) => {
+    if (!currentTarget.value) {
```

Comment:
should we consider a case for too few letters in search term? ('a' or '0' for example will likely match everything)